### ISSUE TITLE
QueryTypes: ToView() on two query types in the same hierarchy (mapping to different views) tries to map each type to the different view

### ISSUE THREAD

**Issue by maumar** [aspnet/EntityFrameworkCore#11808]
Repro:

```cs
    class Program
    {
        static void Main(string[] args)
        {
            using (var ctx = new MyContext())
            {
                ctx.Database.EnsureDeleted();
                ctx.Database.EnsureCreated();

                var f1 = new Faction { Name = "Skeliege" };
                var f2 = new Faction { Name = "Monsters" };
                var f3 = new Faction { Name = "Nilfgaard" };
                var f4 = new Faction { Name = "Northern Realms" };
                var f5 = new Faction { Name = "Scioia'tael" };

                var l11 = new Leader { Faction = f1, Name = "Bran Tuirseach" };
                var l12 = new Leader { Faction = f1, Name = "Crach an Craite" };
                var l13 = new Leader { Faction = f1, Name = "Eist Tuirseach" };
                var l14 = new Leader { Faction = f1, Name = "Harald the Cripple" };

                ctx.Factions.AddRange(f1, f2, f3, f4, f5);
                ctx.Leaders.AddRange(l11, l12, l13, l14);
                ctx.SaveChanges();
            }

            using (var ctx = new MyContext())
            {
                var q1 = ctx.Set<LeaderBase>().ToList(); // queries LeaderBases table
                var q2 = ctx.Set<Leader>().ToList(); // also queries LeaderBases table

                var q3 = ctx.Query<LeaderQueryBase>().ToList(); // queries LeaderBasesView
                var q4 = ctx.Query<LeaderQuery>().ToList(); // queries LeadersView
            }
        }
    }

    public class MyContext : DbContext
    {
        public DbSet<Faction> Factions { get; set; }
        public DbSet<LeaderBase> Leaders { get; set; }

        public DbQuery<LeaderQueryBase> LeadersQuery { get; set; }
        public DbQuery<FactionQuery> FactionsQuery { get; set; }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            optionsBuilder.UseSqlServer(@"Server=.;Database=QueryTypesRepro;Trusted_Connection=True;MultipleActiveResultSets=True");
        }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<LeaderBase>().ToTable("LeaderBases");
            modelBuilder.Entity<Leader>().ToTable("Leaders");

            modelBuilder.Query<LeaderQueryBase>().ToView("LeaderBasesView");
            modelBuilder.Query<LeaderQuery>().ToView("LeadersView");
        }
    }

    public class Faction
    {
        public int Id { get; set; }
        public string Name { get; set; }

        public List<Leader> Leaders { get; set; }
    }

    public class FactionQuery
    {
        public string Name { get; set; }
    }

    public class Leader : LeaderBase
    {
        public Faction Faction { get; set; }
    }

    public class LeaderBase
    {
        public int Id { get; set; }
        public string Name { get; set; }
    }

    public class LeaderQuery
    {
        public Faction Faction { get; set; }
    }

    public class LeaderQueryBase
    {
        public string Name { get; set; }
    }
```

**Comment by ajcvickers** [comment]
Triage: we will make this a validation error in 2.1.

**Comment by ajcvickers** [comment]
This is approved for RTM, but please wait for the branch to open for RTM before merging.

**Comment by AndriySvyryd** [comment]
Fixed in 6be065ceceac92384846f090c2466fbc2ad952fc
